src/main/java/de/retest/recheck/review/ignore/matcher/ElementClassMatcher.java

```diff
@@ -25,12 +25,20 @@ private ElementClassMatcher( final List<String> classValues ) {
 	}
 
 	@Override
-	public boolean test( final Element element ) {
-		final List<String> classValues = toClassValuesList( toClassValueString( element ) );
-		if ( classValues.isEmpty() ) {
+	public boolean test( Element element ) {
+		final List<String> elementClassValues = toClassValuesList( toClassValueString( element ) );
```

Comment:
Less verbose when using recursion:

```java
@Override
public boolean test( Element element ) {
    // test predicate
    final Element parent = element.getParent();
    return parent != null && test( parent );
}
```

And if we want to extract testing parents:

```java
public static boolean testParents( final Element element, final Matcher<Element> matcher ) {
    final Element parent = element.getParent();
    return parent != null && matcher.test( parent );
}
```

Which then can be called from each matcher like:

```java
public boolean test( Element element ) {
    // test predicate
    testParents( element, this );
}
```